### DISCUSSION TITLE
How to setup open telemetry in a remix project with vite?

### DISCUSSION THREAD

**Discussion by gabriel-rye** [remix-run/remix#10169] [Q&A]
Im trying to setup open telemetry with remix.

Looking the the build server folder I see the telemetry code is added right after the imports. But the telemetry code must be before imports to allow it to wrap packages.

Any idea how can I do this?

**Comment by pasunboneleve** [comment]
I'm happy to pair with you on this issue. I'm working my way through https://github.com/mikearnaldi/vite-remix-effect. That project works, but it uses an outdated version of **Effect**, and I'm having trouble understanding the logic so I could reproduce it with the current version.

I **think** **Remix** has no inbuilt way of deciding where the **OpenTelemetry** library will run. In my experience, I get errors in the browser complaining of an initialisation race condition, that is, **OpenTelemetry** tries to initialise twice at every page load. Nasty.

**Comment by pasunboneleve** [comment]
Got it working, in my case for [Honeycomb](https://github.com/remix-run/remix/discussions/9467#discussioncomment-11095527). OpenTelemetry shouln't be much different.
